Source/WebCore/rendering/PathOperation.cpp

```diff
@@ -48,18 +52,32 @@ ReferencePathOperation::ReferencePathOperation(const String& url, const AtomStri
 {
 }
 
+ReferencePathOperation::ReferencePathOperation(std::optional<Path>&& path)
+    : PathOperation(Reference)
+    , m_path(WTFMove(path))
+{
+}
+
 const SVGElement* ReferencePathOperation::element() const
 {
     return m_element.get();
 }
 
-const std::optional<Path> ReferencePathOperation::getPath(const FloatRect&, FloatPoint, OffsetRotation) const
+const std::optional<Path> ReferencePathOperation::path() const
 {
+    if (m_path)
+        return *m_path;
+
     if (!is<SVGPathElement>(m_element.get()) && !is<SVGGeometryElement>(m_element.get()))
         return std::nullopt;
     return pathFromGraphicsElement(m_element.get());
```

Comment:
Since we're adding m_path, do we want to cache the result of this for if we query the path twice?

Reply:
@nmoucht can we compute the path upfront?

Reply:
For ReferencePathOperation yes it can since it is just computed based off of the SVGElement.

Reply:
I've updated the patch to set `m_path` in the constructor so that it's computed upfront.